lib/windows/launcher/actions/appsActions.js

```diff
@@ -199,6 +203,36 @@ export function hideConfirmLaunchDialogAction() {
     };
 }
 
+export function setAppIconPath(source, name, iconPath) {
+    return {
+        type: SET_APP_ICON_PATH,
+        source,
+        name,
+        iconPath,
+    };
+}
+
+/**
+ * Download app icon and dispatch icon update event
+ *
+ * @param {string} source app source identifier
+ * @param {string} name app name
+ * @param {string} iconUrl url to download from
+ * @returns {void}
+ */
+function downloadAppIcon(source, name, iconUrl) {
+    return dispatch => {
+        const iconPath = `${config.getAppsRootDir(source)}/${name}.svg`;
```

Comment:
Maybe I should make iconPath a parameter. What do you think?

Reply:
Yes you could, also a bit further down you could use more destrcturing.